### PR TITLE
LINK-1812 | Button to load ics file

### PR DESCRIPTION
**Pull request by jorilindell**
## Description :sparkles:
Downloading ics file from the signup completed page

## Closes
[LINK-1812](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1812)

# Screenshot
Screenshot 2024-02-08 at 16 47 43

[LINK-1812]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1812?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**LINKEDREGISTRATIONS-UI branch is deployed to platta: https://linkedregistrations-pr137-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDREGISTRATIONS-UI branch is deployed to platta: https://linkedregistrations-pr137-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDREGISTRATIONS-UI branch is deployed to platta: https://linkedregistrations-pr137-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedregistrations-pr137-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/event/eventCalendarButton/EventCalendarButton.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  const locale = useLocale();
+  const { t } = useTranslation('common');
+  const { addNotification } = useNotificationsContext();
+
+  return (
+    <Button
+      {...rest}
+      iconLeft={<IconCalendar aria-hidden />}
+      onClick={() =>
+        downloadEventIcsFile({ addNotification, event, locale, t })
```
Comment (harriris-vincit) [inline src/domain/event/eventCalendarButton/EventCalendarButton.tsx:29]: Maybe the download function could be passed as a prop like variant to allow using a different function too?

File: src/domain/event/utils.ts [hunk 2]
```diff
@@ -18,3 +24,94 @@ export const getEventFields = (event: Event, locale: Language): EventFields => {
...
+      .filter((e) => e)
+      .join(', ') || '-'
+  );
+};
+
+const getDateArray = (date: Date): DateArray => {
+  const parts = formatDate(date, 'yyyy-M-d-H-m')
+    .split('-')
+    .map((e) => Number(e));
+  return [parts[0], parts[1], parts[2], parts[3], parts[4]];
```
Comment (harriris-vincit) [inline src/domain/event/utils.ts:55]:
```suggestion
  return [date.getFullYear(), date.getMonth() + 1, date.getDate(), date.getHours(), date.getMinutes()];
```
Possibly gives the same result?

File: src/domain/event/eventCalendarButton/EventCalendarButton.tsx (hunk 1 above)
Comment (jorilindell) [reply src/domain/event/eventCalendarButton/EventCalendarButton.tsx]: Added support for custom onClick function.

File: src/domain/event/utils.ts (hunk 2 above)
Comment (jorilindell) [reply src/domain/event/utils.ts]: I was actually thinking to implement it like that but didn't the `+ 1 ` part when defining months 😄 
Anyway, changed as suggested.

**Comment by terovirtanen** [comment]
**LINKEDREGISTRATIONS-UI branch is deployed to platta: https://linkedregistrations-pr137-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedregistrations-pr137-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Review by harriris-vincit — APPROVED** [review]
LGTM 🌟

**Pull request merged**